tests/functional/cli/test_cli_repository.py

```diff
@@ -45,6 +49,49 @@ def test_list_all_commits(gitlab_cli, project):
     assert len(ret_all.stdout) > len(ret.stdout)
 
 
+def test_commit_merge_requests(gitlab_cli, project, merge_request, wait_for_sidekiq):
+    """This tests the `project-commit merge-requests` command and also tests
+    that we can print the result using the `json` formatter"""
+    # create and then merge a merge-request
+    source_branch = "test_commit_merge_requests"
+    mr = merge_request(source_branch=source_branch)
+    merge_result = mr.merge(should_remove_source_branch=True)
+    result = wait_for_sidekiq(timeout=60)
+    assert result is True, "sidekiq process should have terminated but did not"
```

Comment:
I see we are adding this assert in all the tests - maybe we should just put this in the fixture itself to avoid repetition? Or is that too strict.

Reply:
Well consumers of the fixture don't necessarily have to merge. And it is doing additional things before we would hand things back to the cleanup part of the fixture.

So in this case would still want to have sidekiq finish (to be safe). So I'm not sure how to move it to the fixture...

Reply:
One idea could be to add a new argument to `wait_for_sidekiq`. Maybe `wait_for_sidekiq(timeout=60, fail_if_timeout=True)` ?

Reply:
Yes, I think the way it was done in #2297 looks nice :)

Reply:
> Yes, I think the way it was done in #2297 looks nice :)

Done